modRlp/src/org/aion/rlp/CompactEncoder.java

```diff
@@ -31,56 +31,54 @@
  *     Samuel Neves through the BLAKE2 implementation.
  *     Zcash project team.
  *     Bitcoinj team.
- ******************************************************************************/
+ */
 package org.aion.rlp;
 
 import static java.util.Arrays.copyOf;
 import static java.util.Arrays.copyOfRange;
 import static org.aion.base.util.ByteUtil.appendByte;
-import static org.aion.rlp.Utils.concatenate;
 import static org.aion.rlp.Utils.hexEncode;
 
 import java.io.ByteArrayOutputStream;
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
 
 /**
  * Compact encoding of hex sequence with optional terminator
  *
- * The traditional compact way of encoding a hex string is to convert it into
- * binary - that is, a string like 0f1248 would become three bytes 15, 18, 72.
- * However, this approach has one slight problem: what if the length of the hex
- * string is odd? In that case, there is no way to distinguish between, say,
- * 0f1248 and f1248.
+ * <p>The traditional compact way of encoding a hex string is to convert it into binary - that is, a
+ * string like 0f1248 would become three bytes 15, 18, 72. However, this approach has one slight
+ * problem: what if the length of the hex string is odd? In that case, there is no way to
+ * distinguish between, say, 0f1248 and f1248.
+ *
+ * <p>Additionally, our application in the Merkle Patricia tree requires the additional feature that
+ * a hex string can also have a special "terminator symbol" at the end (denoted by the 'T'). A
+ * terminator symbol can occur only once, and only at the end.
  *
- * Additionally, our application in the Merkle Patricia tree requires the
- * additional feature that a hex string can also have a special "terminator
- * symbol" at the end (denoted by the 'T'). A terminator symbol can occur only
- * once, and only at the end.
+ * <p>An alternative way of thinking about this to not think of there being a terminator symbol, but
+ * instead treat bit specifying the existence of the terminator symbol as a bit specifying that the
+ * given node encodes a final node, where the value is an actual value, rather than the hash of yet
+ * another node.
  *
- * An alternative way of thinking about this to not think of there being a
- * terminator symbol, but instead treat bit specifying the existence of the
- * terminator symbol as a bit specifying that the given node encodes a final
- * node, where the value is an actual value, rather than the hash of yet another
- * node.
+ * <p>To solve both of these issues, we force the first nibble of the final byte-stream to encode
+ * two flags, specifying oddness of length (ignoring the 'T' symbol) and terminator status; these
+ * are placed, respectively, into the two lowest significant bits of the first nibble. In the case
+ * of an even-length hex string, we must introduce a second nibble (of value zero) to ensure the
```

Comment:
I understand what it's saying, but I find this last sentence quite confusingly worded. If it's not a bother, I'd suggest rephrasing it so that it doesn't sound like "we must ensure that even-length hex strings are even-length hex strings".